### ISSUE TITLE
AlexNet finetune on wikiart style- error

### ISSUE THREAD

**Issue by rahimentezari** [BVLC/caffe#3679]
I just wanted to finetune Alexnet on wiki style using this link:
 https://computing.ece.vt.edu/~f15ece6504/homework2/ 
I used this command :
 sudo /home/pgm/Rahim/caffe-master/build/tools/caffe train -solver solver.prototxt -weights /home/pgm/Rahim/caffe-master/models/bvlc_alexnet/bvlc_alexnet.caffemodel 

I got this error:
I0216 13:27:13.279832  7847 caffe.cpp:185] Using GPUs 0
I0216 13:27:13.292620  7847 caffe.cpp:190] GPU 0: GeForce GTX 980 Ti
I0216 13:27:13.467829  7847 solver.cpp:48] Initializing solver from parameters: 
test_iter: 1000
test_interval: 1000
base_lr: 0.001
display: 20
max_iter: 450000
lr_policy: "step"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
stepsize: 10000
snapshot: 10000
snapshot_prefix: "/home/pgm/Rahim/Tutorials/finetune_example/VT-F15-ECE6504-HW2-1.0/2_finetuning-alexnet-wikiart-style/snapshot"
solver_mode: GPU
device_id: 0
net: "/home/pgm/Rahim/caffe-master/models/bvlc_alexnet"
I0216 13:27:13.468070  7847 solver.cpp:91] Creating training net from net file: /home/pgm/Rahim/caffe-master/models/bvlc_alexnet
I0216 13:27:13.468144  7847 net.cpp:49] Initializing net from parameters: 
state {
  phase: TRAIN
}
I0216 13:27:13.468209  7847 net.cpp:283] Network initialization done.
I0216 13:27:13.468258  7847 solver.cpp:181] Creating test net (#0) specified by net file: /home/pgm/Rahim/caffe-master/models/bvlc_alexnet
I0216 13:27:13.468283  7847 net.cpp:49] Initializing net from parameters: 
state {
  phase: TEST
}
I0216 13:27:13.468318  7847 net.cpp:283] Network initialization done.
I0216 13:27:13.468341  7847 solver.cpp:60] Solver scaffolding done.
I0216 13:27:13.468372  7847 caffe.cpp:129] Finetuning from /home/pgm/Rahim/caffe-master/models/bvlc_alexnet/bvlc_alexnet.caffemodel
I0216 13:27:14.012931  7847 upgrade_proto.cpp:42] Attempting to upgrade input file specified using deprecated transformation parameters: /home/pgm/Rahim/caffe-master/models/bvlc_alexnet/bvlc_alexnet.caffemodel
I0216 13:27:14.012982  7847 upgrade_proto.cpp:45] Successfully upgraded file specified using deprecated data transformation parameters.
W0216 13:27:14.012995  7847 upgrade_proto.cpp:47] Note that future Caffe releases will only support transform_param messages for transformation fields.
I0216 13:27:14.013124  7847 upgrade_proto.cpp:51] Attempting to upgrade input file specified using deprecated V1LayerParameter: /home/pgm/Rahim/caffe-master/models/bvlc_alexnet/bvlc_alexnet.caffemodel
I0216 13:27:14.242383  7847 upgrade_proto.cpp:59] Successfully upgraded file specified using deprecated V1LayerParameter
I0216 13:27:14.258903  7847 net.cpp:816] Ignoring source layer data
I0216 13:27:14.258942  7847 net.cpp:816] Ignoring source layer conv1
I0216 13:27:14.258952  7847 net.cpp:816] Ignoring source layer relu1
I0216 13:27:14.258961  7847 net.cpp:816] Ignoring source layer norm1
I0216 13:27:14.258970  7847 net.cpp:816] Ignoring source layer pool1
I0216 13:27:14.258980  7847 net.cpp:816] Ignoring source layer conv2
I0216 13:27:14.258987  7847 net.cpp:816] Ignoring source layer relu2
I0216 13:27:14.258996  7847 net.cpp:816] Ignoring source layer norm2
I0216 13:27:14.259006  7847 net.cpp:816] Ignoring source layer pool2
I0216 13:27:14.259014  7847 net.cpp:816] Ignoring source layer conv3
I0216 13:27:14.259023  7847 net.cpp:816] Ignoring source layer relu3
I0216 13:27:14.259032  7847 net.cpp:816] Ignoring source layer conv4
I0216 13:27:14.259040  7847 net.cpp:816] Ignoring source layer relu4
I0216 13:27:14.259049  7847 net.cpp:816] Ignoring source layer conv5
I0216 13:27:14.259057  7847 net.cpp:816] Ignoring source layer relu5
I0216 13:27:14.259066  7847 net.cpp:816] Ignoring source layer pool5
I0216 13:27:14.259075  7847 net.cpp:816] Ignoring source layer fc6
I0216 13:27:14.259083  7847 net.cpp:816] Ignoring source layer relu6
I0216 13:27:14.259093  7847 net.cpp:816] Ignoring source layer drop6
I0216 13:27:14.259101  7847 net.cpp:816] Ignoring source layer fc7
I0216 13:27:14.259109  7847 net.cpp:816] Ignoring source layer relu7
I0216 13:27:14.259119  7847 net.cpp:816] Ignoring source layer drop7
I0216 13:27:14.259126  7847 net.cpp:816] Ignoring source layer fc8
I0216 13:27:14.259135  7847 net.cpp:816] Ignoring source layer loss
I0216 13:27:14.828289  7847 upgrade_proto.cpp:42] Attempting to upgrade input file specified using deprecated transformation parameters: /home/pgm/Rahim/caffe-master/models/bvlc_alexnet/bvlc_alexnet.caffemodel
I0216 13:27:14.828327  7847 upgrade_proto.cpp:45] Successfully upgraded file specified using deprecated data transformation parameters.
W0216 13:27:14.828368  7847 upgrade_proto.cpp:47] Note that future Caffe releases will only support transform_param messages for transformation fields.
I0216 13:27:14.828384  7847 upgrade_proto.cpp:51] Attempting to upgrade input file specified using deprecated V1LayerParameter: /home/pgm/Rahim/caffe-master/models/bvlc_alexnet/bvlc_alexnet.caffemodel
I0216 13:27:15.061686  7847 upgrade_proto.cpp:59] Successfully upgraded file specified using deprecated V1LayerParameter
I0216 13:27:15.076108  7847 net.cpp:816] Ignoring source layer data
I0216 13:27:15.076155  7847 net.cpp:816] Ignoring source layer conv1
I0216 13:27:15.076167  7847 net.cpp:816] Ignoring source layer relu1
I0216 13:27:15.076176  7847 net.cpp:816] Ignoring source layer norm1
I0216 13:27:15.076185  7847 net.cpp:816] Ignoring source layer pool1
I0216 13:27:15.076195  7847 net.cpp:816] Ignoring source layer conv2
I0216 13:27:15.076203  7847 net.cpp:816] Ignoring source layer relu2
I0216 13:27:15.076212  7847 net.cpp:816] Ignoring source layer norm2
I0216 13:27:15.076222  7847 net.cpp:816] Ignoring source layer pool2
I0216 13:27:15.076231  7847 net.cpp:816] Ignoring source layer conv3
I0216 13:27:15.076241  7847 net.cpp:816] Ignoring source layer relu3
I0216 13:27:15.076249  7847 net.cpp:816] Ignoring source layer conv4
I0216 13:27:15.076258  7847 net.cpp:816] Ignoring source layer relu4
I0216 13:27:15.076267  7847 net.cpp:816] Ignoring source layer conv5
I0216 13:27:15.076275  7847 net.cpp:816] Ignoring source layer relu5
I0216 13:27:15.076284  7847 net.cpp:816] Ignoring source layer pool5
I0216 13:27:15.076293  7847 net.cpp:816] Ignoring source layer fc6
I0216 13:27:15.076303  7847 net.cpp:816] Ignoring source layer relu6
I0216 13:27:15.076310  7847 net.cpp:816] Ignoring source layer drop6
I0216 13:27:15.076319  7847 net.cpp:816] Ignoring source layer fc7
I0216 13:27:15.076328  7847 net.cpp:816] Ignoring source layer relu7
I0216 13:27:15.076336  7847 net.cpp:816] Ignoring source layer drop7
I0216 13:27:15.076345  7847 net.cpp:816] Ignoring source layer fc8
I0216 13:27:15.076354  7847 net.cpp:816] Ignoring source layer loss
I0216 13:27:15.106338  7847 caffe.cpp:219] Starting Optimization
I0216 13:27:15.106392  7847 solver.cpp:280] Solving 
I0216 13:27:15.106410  7847 solver.cpp:281] Learning Rate Policy: step
I0216 13:27:15.106432  7847 solver.cpp:338] Iteration 0, Testing net (#0)
F0216 13:27:15.106477  7847 net.cpp:567] Check failed: end < layers_.size() (-1 vs. 0) 
*** Check failure stack trace: ***
    @     0x7f672cb9fdaa  (unknown)
    @     0x7f672cb9fce4  (unknown)
    @     0x7f672cb9f6e6  (unknown)
    @     0x7f672cba2687  (unknown)
    @     0x7f6732324abd  caffe::Net<>::ForwardFromTo()
    @     0x7f6732324d0f  caffe::Net<>::ForwardPrefilled()
    @     0x7f673230ab64  caffe::Solver<>::Test()
    @     0x7f673230b71d  caffe::Solver<>::TestAll()
    @     0x7f6732312335  caffe::Solver<>::Step()
    @     0x7f6732312f62  caffe::Solver<>::Solve()
    @           0x4097e8  train()
    @           0x406783  main
    @     0x7f6725511ec5  (unknown)
    @           0x405cd9  (unknown)
    @              (nil)  (unknown)

**Comment by seanbell** [comment]
You have 0 layers after loading.  You can see that `layers_.size()` is 0.

Your solver is referencing `/home/pgm/Rahim/caffe-master/models/bvlc_alexnet` -- is that a directory?  You need to reference the actual prototxt file that contains all the layers etc.

Or, Is it possible that you mixed `layer {` and `layers {` in one of the prototxt files?

Also, never use `sudo` with caffe.  Fix your permissions and then don't use `sudo`.

**Comment by shelhamer** [comment]
From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
